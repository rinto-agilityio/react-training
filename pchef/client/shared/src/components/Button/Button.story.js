// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

// Helper
import { wInfo } from '../../../.storybook/utils';

// Component
import Button from './index'

storiesOf('Button', module)
  .addDecorator(wInfo())
  .add('Default', () => (
    <Button
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Success', () => (
    <Button
      type="success"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Danger', () => (
    <Button
      type="danger"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Warning', () => (
    <Button
      type="warning"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Info', () => (
    <Button
      type="info"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))
