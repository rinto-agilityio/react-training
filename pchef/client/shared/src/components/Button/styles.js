// Lib
import { StyleSheet } from 'react-native'

// Themes
import { COLORS, METRICS, FONTS } from '../../themes'

// Create style for button
const styles = StyleSheet.create({
  button: {
    ...METRICS.flexCenter,
    padding: METRICS.padding.sm,
  },
  title: {
    color: COLORS.white,
    textTransform: 'uppercase',
    fontWeight: FONTS.fontWeight.large,
  },
  default: {
    backgroundColor: COLORS.baseBlue,
  },
  success: {
    backgroundColor: COLORS.green,
  },
  danger: {
    backgroundColor: COLORS.red,
  },
  warning: {
    backgroundColor: COLORS.yellow,
  },
  info: {
    backgroundColor: COLORS.darkWhite,
  },
})

export default styles
