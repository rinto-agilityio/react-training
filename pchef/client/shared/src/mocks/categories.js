export const categories = [
  {
    id: 'chicken',
    name: 'Chicken',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Heathy',
    name: 'Heathy',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Easy Breakast',
    name: 'Easy Breakast',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Dessert',
    name: 'Dessert',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Fish',
    name: 'Fish',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Pizza',
    name: 'Pizza',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Cake',
    name: 'Cake',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Deserts',
    name: 'Deserts',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Salad',
    name: 'Salad',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Kids',
    name: 'Kids',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Meat',
    name: 'Meat',
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
  {
    id: 'Easy Lunch',
    name: null,
    imgUrl:
      'https://realhousemoms.com/wp-content/uploads/Honey-Butter-Chicken-IG.jpg',
  },
]
