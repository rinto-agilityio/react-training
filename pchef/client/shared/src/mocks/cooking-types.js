export const cookingTypes = [
  {
    id: 'baking',
    name: 'Baking',
  },
  {
    id: 'roasting',
    name: 'Roasting',
  },
  {
    id: 'frying',
    name: 'Frying',
  },
]
