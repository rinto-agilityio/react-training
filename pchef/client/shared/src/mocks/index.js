// mock categories
export const category = {
  name: 'Heathy',
  imgUrl:
    'https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500',
}
