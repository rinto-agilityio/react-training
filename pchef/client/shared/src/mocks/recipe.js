export const recipe = {
  description: '2lb pork neck bones, 2lb beef bones, 1lb oxtail',
  title: 'Bun Bo Hue Recipe',
  imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
  votes: [1],
  steps: [
    {
      step: 1,
      description: 'Step 1 description',
      title: 'Step 1 title',
      subTitle: 'Step 1 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 2,
      description: 'Step 2 description',
      title: 'Step 2 title',
      subTitle: 'Step 2 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 3,
      description: 'Step 3 description',
      title: 'Step 3 title',
      subTitle: 'Step 3 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 4,
      description: 'Step 4 description',
      title: 'Step 4 title',
      subTitle: 'Step 4 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 5,
      description: 'Step 5 description',
      title: 'Step 5 title',
      subTitle: 'Step 5 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
    {
      step: 6,
      description: 'Step 6 description',
      title: 'Step 6 title',
      subTitle: 'Step 6 sub title',
      imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    },
  ],
}
