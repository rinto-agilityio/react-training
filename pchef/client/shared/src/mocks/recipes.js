export const recipes = [
  {
    id: 0,
    title: 'Bun Bo Hue Recipe',
    subTitle: 'Traditional food',
    description: '2lb pork neck bones, 2lb beef bones, 1lb oxtail',
    imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    votes: [1],
    views: 1,
    steps: [
      {
        step: 1,
        description: 'Step 1 description',
        title: 'Step 1 title',
        subTitle: 'Step 1 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 2,
        description: 'Step 2 description',
        title: 'Step 2 title',
        subTitle: 'Step 2 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 3,
        description: 'Step 3 description',
        title: 'Step 3 title',
        subTitle: 'Step 3 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 4,
        description: 'Step 4 description',
        title: 'Step 4 title',
        subTitle: 'Step 4 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 5,
        description: 'Step 5 description',
        title: 'Step 5 title',
        subTitle: 'Step 5 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
      {
        step: 6,
        description: 'Step 6 description',
        title: 'Step 6 title',
        subTitle: 'Step 6 sub title',
        imgUrl: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
      },
    ],
  },
  {
    id: 1,
    title: 'Crispy Fried Chicken',
    imgUrl:
      'https://www.tasteofhome.com/wp-content/uploads/2018/01/Crispy-Fried-Chicken_EXPS_FRBZ19_6445_C01_31_3b.jpg',

    description: 'Description for Crispy Fried Chicken',
  },
  {
    id: 2,
    title: 'Crispy Fried Chicken',
    imgUrl:
      'https://www.tasteofhome.com/wp-content/uploads/2018/01/Crispy-Fried-Chicken_EXPS_FRBZ19_6445_C01_31_3b.jpg',

    description: 'Description for Crispy Fried Chicken',
  },
  {
    id: 3,
    title: 'Crispy Fried Chicken',
    imgUrl:
      'https://www.tasteofhome.com/wp-content/uploads/2018/01/Crispy-Fried-Chicken_EXPS_FRBZ19_6445_C01_31_3b.jpg',

    description: 'Description for Crispy Fried Chicken',
  },
]
