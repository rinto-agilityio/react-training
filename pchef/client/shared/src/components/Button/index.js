// libs
import React from 'react'
import { Button as ButtonComponent, Text } from 'react-native-paper'

// Styles
import styles from './styles'
import { COLORS } from '../../themes/index'

type Props = {
  onPress?: () => void,
  buttonStyle?: {} | Array<{}>,
  title: string,
  titleStyle?: {} | Array<{}>,
  disabled?: boolean,
  mode?: string,
  contentStyle?: {} | Array<{}>,
  color?: string,
  loading?: boolean,
  type?: string,
}

const Button = ({
  onPress,
  buttonStyle,
  title,
  mode,
  type,
  disabled,
  contentStyle,
  color,
  loading,
  titleStyle,
}: Props) => (
  <ButtonComponent
    color={color}
    loading={loading}
    onPress={onPress}
    mode={mode}
    disabled={disabled}
    style={[styles.button, styles[type], buttonStyle]}
    contentStyle={[styles.content, contentStyle]}
  >
    <Text style={[styles.title, titleStyle]}>{title}</Text>
  </ButtonComponent>
)

Button.defaultProps = {
  onPress: () => {},
  buttonStyle: {},
  titleStyle: {},
  disabled: false,
  mode: 'contained',
  contentStyle: {},
  color: COLORS.baseBlue,
  loading: false,
  type: 'default',
}

export default Button
