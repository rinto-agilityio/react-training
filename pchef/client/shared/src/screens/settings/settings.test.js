import React from 'react'

// Mock
import { user } from '../../mocks'

// Components
import Settings from '.'
import Button from '../../components/Button'

const props = {
  user,
  handleLogout: jest.fn(),
}

describe('Screens', () => {
  describe('Settings', () => {
    it('should render Setting screen', () => {
      const renderComponent = renderer.create(<Settings {...props} />)
      expect(renderComponent).toMatchSnapshot()
    })

    it('should call handleLogout when click logout button', () => {
      props.handleLogout.mockReturnValue('Icon on press invoked')

      const wrapper = shallow(<Settings {...props} />)

      wrapper
        .find(Button)
        .props()
        .onPress()
      expect(props.handleLogout.mock.calls.length).toBe(1)
    })
  })
})
