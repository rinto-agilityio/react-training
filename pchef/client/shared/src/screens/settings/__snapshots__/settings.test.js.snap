// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screens Settings should render Setting screen 1`] = `
<div
  className="css-view-1dbjc4n r-padding-1sp51qo r-width-13qz1uu"
>
  <div
    className="css-text-901oao r-color-1h2fve2"
    dir="auto"
  >
    User name:
  </div>
  <input
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect="on"
    className="css-textinput-1cwyjr8 r-borderBottomWidth-qklmqi r-borderColor-34efmp r-borderStyle-1phboty r-color-cqee49 r-fontSize-1i10wst r-height-1g7fiml r-marginBottom-1yflyrw r-marginTop-156q2ks"
    data-focusable={true}
    dir="auto"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onSelect={[Function]}
    readOnly={true}
    spellCheck={true}
    type="text"
    value="Alex"
  />
  <div
    className="css-text-901oao r-color-1h2fve2"
    dir="auto"
  >
    Email:
  </div>
  <input
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect="on"
    className="css-textinput-1cwyjr8 r-borderBottomWidth-qklmqi r-borderColor-34efmp r-borderStyle-1phboty r-color-cqee49 r-fontSize-1i10wst r-height-1g7fiml r-marginBottom-1yflyrw r-marginTop-156q2ks"
    data-focusable={true}
    dir="auto"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onSelect={[Function]}
    readOnly={true}
    spellCheck={true}
    type="text"
    value="alex@gmail.com"
  />
  <div
    className="css-view-1dbjc4n r-borderStyle-1phboty"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxPack": "center",
        "WebkitJustifyContent": "center",
        "alignItems": "center",
        "backgroundColor": "rgba(211,43,83,1.00)",
        "borderBottomColor": "rgba(0,0,0,0.00)",
        "borderBottomLeftRadius": "4px",
        "borderBottomRightRadius": "4px",
        "borderBottomWidth": "0px",
        "borderLeftColor": "rgba(0,0,0,0.00)",
        "borderLeftWidth": "0px",
        "borderRightColor": "rgba(0,0,0,0.00)",
        "borderRightWidth": "0px",
        "borderTopColor": "rgba(0,0,0,0.00)",
        "borderTopLeftRadius": "4px",
        "borderTopRightRadius": "4px",
        "borderTopWidth": "0px",
        "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
        "justifyContent": "center",
        "marginTop": "30px",
        "minWidth": "64px",
        "msFlexAlign": "center",
        "msFlexPack": "center",
        "paddingBottom": "5px",
        "paddingLeft": "5px",
        "paddingRight": "5px",
        "paddingTop": "5px",
      }
    }
  >
    <div
      className="css-cursor-18t94o4 css-view-1dbjc4n r-backgroundColor-1niwhzg r-cursor-1loqt21 r-touchAction-1otgn73 r-userSelect-lrvibr"
      data-focusable={true}
      disabled={false}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      style={
        Object {
          "borderBottomLeftRadius": "4px",
          "borderBottomRightRadius": "4px",
          "borderTopLeftRadius": "4px",
          "borderTopRightRadius": "4px",
          "overflowX": "hidden",
          "overflowY": "hidden",
        }
      }
      tabIndex="0"
    >
      <div
        className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1777fci"
      >
        <div
          className="css-text-901oao css-textOneLine-bfa6kz r-color-cqee49 r-letterSpacing-ud0q2t r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr r-textAlign-q4m81j"
          dir="auto"
          style={
            Object {
              "direction": "ltr",
              "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            }
          }
        >
          <span
            className="css-text-901oao css-textHasAncestor-16my406 r-color-jwli3a r-fontWeight-1kfrs79 r-textTransform-tsynxw"
            dir="auto"
            style={
              Object {
                "direction": "ltr",
                "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "textAlign": "left",
              }
            }
          >
            Logout
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
